framework/src/task_manager/taskmgr_start.c

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdio.h>
+#include <sys/types.h>
+#include <apps/builtin.h>
+#include <tinyara/taskmgt.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * taskmgr_start
+ ****************************************************************************/
+int taskmgr_start(int handle, int caller_pid)
+{
+	int ret;
+
+	if (IS_INVALID_HANDLE(handle)) {
+		return TM_INVALID_PARAM;
+	}
+
+	tm_update_task_state(handle);
+	ret = tm_get_task_state(handle);
+	if (ret != TM_TASK_STATE_STOP) {
+		ret = -ret;
+		return ret;
+	}
+
+	/* handle is for starting task, pid is for caller */
+	ret = tm_check_permission(handle, caller_pid);
+	if (ret != OK) {
+		return TM_FAIL_NOT_PERMITTED;
```

Comment:
As far as I know, Jeongchan changed ```tm_check_permission``` to return TM_FAIL_NOT_PERMITTED.
```
return ret;
```
is better